code/modules/atmospherics/machinery/components/unary_devices/thermomachine.dm

```diff
@@ -52,6 +52,16 @@
 	if(board)
 		piping_layer = board.pipe_layer
 		set_layer = piping_layer
+
+	var/turf/turf = get_turf(src)
+	var/obj/machinery/atmospherics/device
+	for(device in turf.contents)
```

Comment:
A: This should just be `for(var/obj/machinery/atmospherics/device as anything in turf)`, you shouldn't need to define this separately, and you don't need to do .contents. Hell you can just iterate over `get_turf(src)`, rather then making a new var for it.
B: What happens when I try and place a pipe under one of these machines? Does it fail as it should?

Reply:
changed it to the final form `for(var/obj/machinery/atmospherics/device as anything in get_turf(src))`, had to remove the error message because the user is not specified

if you try and place a pipe under the machine after is built you get an error message and the action fails

Reply:
Are you sure this should be `as anything`? What if there's something else on that turf?

Reply:
Ah wait I have brainworms yeah, the as anything is bad

Reply:
removed